src/tools/tests/manifest2proto-test.ts

```diff
@@ -159,6 +161,145 @@ describe('manifest2proto', () => {
     });
   });
 
+  it('encodes entity type with simple refinement', async () => {
+    const location = {
+      start: {offset: 0, line: 0, column: 0},
+      end: {offset: 0, line: 0, column: 0}
+    };
+    const refinement = Refinement.fromAst(
+      {
+        kind: 'refinement',
+        expression: {
+          kind: 'boolean-node',
+          value: true,
+          location,
+        },
+        location,
+      }, {});
+    const entity = EntityType.make(
+      ['Foo'],
+      {value: 'Text'},
+      {refinement}
+    );
+    assert.deepStrictEqual(await toProtoAndBackType(entity), {
+      entity: {
+        schema: {
+          names: ['Foo'],
+          fields: {
+            value: {
+              primitive: 'TEXT'
+            }
+          },
+          hash: '1c9b8f8d51ff6e11235ac13bf0c5ca74c88537e0',
+        },
+      },
+      refinement: {
+        boolean: true
+      },
+    });
+  });
+
+  it('encodes entity type with nested refinement', async () => {
+    const manifestAst = parse(`
+        particle Foo
+            input: reads Something {num: Number} [num / 2 < 6 and num  > -1]
+    `);
+    const refinement = Refinement.fromAst(manifestAst[0].args[0].type.refinement, {'num': 'Number'});
+    const entity = EntityType.make(
```

Comment:
Again, why not write a recipe / particle in the manifest and pull out what you want to test? Using these low level APIs is prone to bugs and may require changes across more code if these internal detail get refactored.